src/thor/triplegbuilder.cc

```diff
@@ -1124,10 +1129,11 @@ void TripLegBuilder::Build(
 
     // have to always compute the offset in case the timezone changes along the path
     // we could cache the timezone and just add seconds when the timezone doesnt change
-    time_info = time_info.forward(trip_path.node_size() == 0
-                                      ? 0.0
-                                      : trip_path.node().rbegin()->cost().elapsed_cost().seconds(),
-                                  node->timezone());
+    const auto offset_time =
+        time_info.forward(trip_path.node_size() == 0
+                              ? 0.0
+                              : trip_path.node().rbegin()->cost().elapsed_cost().seconds(),
+                          static_cast<int>(node->timezone()));
```

Comment:
i guess this is a bug i introduced in the `invariant` PR. 

```suggestion
    const auto offset_time =
        time_info.forward(trip_path.node_size() == 0 || invariant
                              ? 0.0
                              : trip_path.node().rbegin()->cost().elapsed_cost().seconds(),
                          static_cast<int>(node->timezone()));
```

Reply:
good catch! but I thinks your suggestion also contains a bug: why do you use such `trip_path.node().rbegin()->cost().elapsed_cost().seconds()` offset ? If we update `time_info` at each iteration we should move forward for time difference between 2 last nodes;
I think, we have similar bug here https://github.com/valhalla/valhalla/blob/master/src/thor/bidirectional_astar.cc#L167-L168 - pass `offset_time` to the method that expects time info for the start of the route

Reply:
hm, just realized, If we update `time_info` on each iteration, it will look like:
```c++
    float seconds_diff = (is_first_edge || invariant)
                         ? 0.f
                         : std::prev(edge_itr)->elapsed_cost.secs;
    if (std::distance(path_begin, edge_itr) > 1)
      seconds_diff -= (edge_itr - 2)->elapsed_cost.secs;
    time_info = time_info.forward(seconds_diff, static_cast<int>(node->timezone()));
```
here `time_info` represents actual time for the current node. If we updated `time_info` on each iteration, we would accumulate rounding error for the time, due to the fact, that `time_info` keeps time as integer;

and, one more problem I noticed: here https://github.com/valhalla/valhalla/blob/master/src/thor/triplegbuilder.cc#L1129 we use offset from the previous node, why ? looks like a bug